src/backend/utils/adt/ruleutils.c

```diff
@@ -10867,7 +10867,7 @@ write_out_rule(PartitionRule *rule, PartitionNode *pn, Node *start,
 {
 	char *str;
 
-	if (!*first_rule)
+	if (first_rule && !*first_rule)
```

Comment:
I think the error is 6c12827097e95912476267d73f3f56e8250b13aad37ab4286c3c7fdab7f30ec7R10896 and should be like this:
```c
		if (strlen(body->buf->data) && !*first_rule &&
```

Reply:
Take a look, `first_rule` isn't dereferenced here, you changed the line in your example
https://github.com/arenadata/gpdb/blob/604bb5312052f3a1d8b62eb38ff3a51231fcd91d/src/backend/utils/adt/ruleutils.c#L10896

Reply:
> Take a look, `first_rule` isn't dereferenced here, you changed the line in your example

Yes, I think first_rule should dereferenced here

Reply:
Then it would be easier just to assert that `first_rule` is not null

Reply:
I think there is a semantic error here: we don't want to print a space before the first rule.
https://github.com/arenadata/gpdb/blob/604bb5312052f3a1d8b62eb38ff3a51231fcd91d/src/backend/utils/adt/ruleutils.c#L10896-L10898

Reply:
Added a condition

Reply:
first_rule should not be NULL

Reply:
write_out_rule is a static function and it is always called with not-NULL first_rule pointer

Reply:
Added assertion and removed checks